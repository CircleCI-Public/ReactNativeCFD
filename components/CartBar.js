import * as React from 'react';
import {Appbar, Paragraph} from 'react-native-paper';
import {StyleSheet} from 'react-native';
import {useEffect} from 'react';

const styles = StyleSheet.create({
  bottom: {
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
  },
  Price: {
    color: 'white',
    fontSize: 20,
    fontFamily: 'Roboto',
    marginLeft: '70%',
    position: 'relative',
    top: 2,
  },
});

const total = cart => {
  if (cart == undefined || cart == [] || cart == null) {
    return 0;
  } else {
    return cart.map(item => item.cost).reduce((prev, curt) => prev + curt);
  }
};

const CartBar = ({cart, makeVisible}) => {
  let tempCart = 0;
  useEffect(() => {
    tempCart = cart;
    console.warn(cart);
  });
  return (
    <Appbar style={styles.bottom}>
      <Appbar.Action
        icon="cart"
        onPress={() => {
          makeVisible();
        }}
        style={styles.ButtonContainer}
      />
      <Paragraph style={styles.Price}>${total(tempCart).toFixed(2)}</Paragraph>
    </Appbar>
  );
};

export default CartBar;
