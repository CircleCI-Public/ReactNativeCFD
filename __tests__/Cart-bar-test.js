import React from 'react';
import renderer from 'react-test-renderer';
import CartBar from '../components/CartBar';

test('renders correctly', () => {
  const tree = renderer.create(<CartBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
