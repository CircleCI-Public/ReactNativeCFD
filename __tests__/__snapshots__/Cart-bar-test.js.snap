// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6200ee",
      "bottom": 0,
      "elevation": 4,
      "flexDirection": "row",
      "height": 56,
      "left": 0,
      "paddingHorizontal": 4,
      "position": "absolute",
      "right": 0,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "height": 36,
            "width": 36,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞Ñê
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "color": "white",
              "fontFamily": "Roboto",
              "fontSize": 20,
              "marginLeft": "70%",
              "position": "relative",
              "top": 2,
            },
          ],
        ],
      ]
    }
  >
    $
    0.00
  </Text>
</View>
`;
